File: src/main/java/com/kmarzecki/communicator/api/board/BoardController.java
package com.kmarzecki.communicator.api.board;

import com.kmarzecki.communicator.service.BoardService;
import lombok.AllArgsConstructor;
import org.springframework.web.bind.annotation.*;

import java.security.Principal;
import java.util.List;
import java.util.UUID;

@RestController
@RequestMapping("/board")
@CrossOrigin(origins = {"*"},allowCredentials = "true", allowedHeaders = "*")
@AllArgsConstructor
public class BoardController {
    BoardService boardService;

    @PutMapping("/{id}")
    BoardResponse update(@RequestBody UpdateBoardRequest request, @PathVariable UUID id, Principal principal) {
        return boardService.update(request, id, principal);
    }

    @PostMapping
    BoardResponse create(@RequestBody  CreateBoardRequest request, Principal principal) {
        return boardService.create(request, principal);
    }

    @GetMapping
    List<BoardResponse> getBoards(Principal principal) {
        return boardService.getBoards(principal);
    }

    @DeleteMapping("/{id}")
    void deleteBoard(@PathVariable  UUID id, Principal principal) {
        boardService.deleteBoard(id, principal);
    }


}

File: src/main/kotlin/com/kmarzecki/trollo/api/board/BoardController.kt
package com.kmarzecki.trollo.api.board

import com.kmarzecki.trollo.service.BoardService
import org.springframework.web.bind.annotation.*
import java.security.Principal
import java.util.*

@RestController
@RequestMapping("/board")
@CrossOrigin(origins = ["*"], allowCredentials = "true", allowedHeaders = ["*"])
class BoardController(
        val boardService: BoardService
) {

    @PutMapping("/{id}")
    fun update(@RequestBody request: UpdateBoardRequest, @PathVariable id: UUID, principal: Principal): BoardResponse? {
        return boardService.update(request, id, principal)
    }

    @PostMapping
    fun create(@RequestBody request: CreateBoardRequest, principal: Principal): BoardResponse? {
        return boardService.create(request, principal)
    }

    @GetMapping
    fun getBoards(principal: Principal): List<BoardResponse> {
        return boardService.getBoards(principal)
    }

    @GetMapping("/{id}/lanes")
    fun getBoard(@PathVariable id: UUID, principal: Principal): List<LaneResponse> {
        return boardService.getLanesForBoard(id, principal)
    }


    @DeleteMapping("/{id}")
    fun deleteBoard(@PathVariable id: UUID, principal: Principal) {
        boardService.deleteBoard(id, principal)
    }
}